_posts/2024-02-27-comet-donation.md

```diff
@@ -0,0 +1,106 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+# Introduction
+The Apache Arrow PMC is pleased to announce the donation of the [Comet project],
+a native Spark SQL Accelerator built on [Apache Arrow DataFusion].
+
+Comet is an Apache Spark plugin that uses Apache Arrow DataFusion to
+accelerate Spark workloads. It is designed as a drop-in
+replacement for Spark's JVM based SQL execution engine and offers significant
+performance improvements for some workloads as shown below.
+
+```text
+   ┌─────────────────────────────────────────────────────────────────┐
+   │                                                                 │
+   │ ┌──────────┐ ┌────────────┐ ┌────────────┐       ┌────────────┐ │
+   │ │   SQL    │ │  Cluster   │ │  DAG/Task  │  ...  │  Executor  │ │
+   │ │ Planner  │ │  Manager   │ │ Scheduler  │       │            │ │
+   │ └──────────┘ └────────────┘ └────────────┘       └────────────┘ │
+   │                                                         │       │
+   └─────────────────────────────────────────────────────────────────┘
+     Spark (JVM Based)                                       │        
+                                  ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─         
+                                                                      
+                                  │                                   
+                                  ▼                                   
+                 ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                   
+Comet Execution  ┃                                ┃                   
+Engine           ┃  ┌─────────────────────────┐   ┃                   
+(Native Code)    ┃  │ Apache Arrow DataFusion │   ┃                   
+                 ┃  └─────────────────────────┘   ┃                   
+                 ┃                                ┃                   
+                 ┃  ┌─────────────────────────┐   ┃                   
+                 ┃  │    Spark Compatible     │   ┃                   
+                 ┃  │  Expressions/Operators  │   ┃                   
+                 ┃  └─────────────────────────┘   ┃                   
+                 ┃                                ┃                   
+                 ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛                   
+```
+
+**Figure 1**: With Comet, users interact with the same Spark ecosystem, tools
+and APIs such as Spark SQL. Queries still run through Spark's mature and feature
+rich query optimizer and planner. However, the execution is delegated to Comet,
+which is significantly faster and more resource efficient than the  JVM based
+implementation.
+
+[Rust]: https://www.rust-lang.org/
+
+# Background
+
+Comet is one of a growing class of projects that aim to accelerate Spark using
+native columnar engines such as the proprietary [Databricks Photon Engine] and
+the open source [Gluten project] and [Spark RAPIDS].
```

Comment:
Would it help to give some background as to why these initiatives have been popping up in the first place? I know the article alludes to columnar execution and lower resource usage, but (imo) it would be more approachable if it were framed from the perspective of the struggles Spark users face **today**, and then explaining how these initiatives address those problems.

Relatedly, I vaguely remember hearing about an effort to bring columnar execution into mainline Spark (was it Project Tungsten?). I would be curious to hear about the differences between that effort and Comet's approach.

Reply:
Thank you @SChakravorti21  for this suggestion. I agree such background would help -- maybe some of the other reviewers (some of whom are spark committers I believe) can offer more specifics and how this relates to mainline spark.

The [Velox Paper](https://vldb.org/pvldb/vol15/p3372-pedreira.pdf) basically says the JVM implementation of spark is slow so native columnar execution is better, but I don't recall it delving into any more detail